lib/galaxy/webapps/galaxy/api/history_contents.py

```diff
@@ -431,16 +431,26 @@ def index_jobs_summary(
         summary="Download the content of a dataset collection as a `zip` archive.",
         response_class=StreamingResponse,
     )
+    def download_history_dataset_collection(
+        self,
+        trans: ProvidesHistoryContext = DependsOnTrans,
+        history_id: EncodedDatabaseIdField = HistoryIDPathParam,
+        id: EncodedDatabaseIdField = HistoryHDCAIDPathParam,
+    ):
+        """Download the content of a history dataset collection as a `zip` archive
+        while maintaining approximate collection structure.
+        """
+        archive = self.service.get_dataset_collection_archive_for_download(trans, id)
+        return StreamingResponse(archive.get_iterator(), headers=archive.get_headers())
+
     @router.get(
-        "/api/dataset_collection/{id}/download",
```

Comment:
This was definitely the problem causing the 404, it was falling back into the legacy API endpoint that doesn't exist anymore. Thanks for catching it!